test/unified_format.py

```diff
@@ -1181,19 +1194,30 @@ def check_events(self, spec):
             events = event_spec["events"]
             # Valid types: 'command', 'cmap'
             event_type = event_spec.get("eventType", "command")
+            ignore_extra_events = event_spec.get("ignoreExtraEvents", False)
+            server_connection_id = event_spec.get("serverConnectionId")
+
             assert event_type in ("command", "cmap")
 
             listener = self.entity_map.get_listener_for_client(client_name)
             actual_events = listener.get_events(event_type)
             if len(events) == 0:
-                self.assertEqual(actual_events, [])
+                if not ignore_extra_events:
+                    self.assertEqual(actual_events, [])
                 continue
 
-            self.assertGreaterEqual(len(actual_events), len(events), actual_events)
+            if ignore_extra_events:
+                self.assertGreaterEqual(len(actual_events), len(events), actual_events)
+            else:
+                self.assertEqual(len(actual_events), len(events), actual_events)
 
             for idx, expected_event in enumerate(events):
                 self.match_evaluator.match_event(event_type, expected_event, actual_events[idx])
 
+            if server_connection_id is not None:
+                assert isinstance(server_connection_id, int)
+                assert server_connection_id < 0
```

Comment:
server_connection_id >= 0?

Reply:
The [spec](https://github.com/mongodb/specifications/blob/223df1f58beb0d3aed1e6ef7f1ee74a707a01a43/source/unified-test-format/unified-test-format.rst#hasserverconnectionid) says it is a "nonpositive Int32"